frontend/src/app/forms/forms.page.ts

```diff
@@ -0,0 +1,175 @@
+
+  constructor(
+    private activeRoute: ActivatedRoute,
+    private navigationController: NavController,
+    private router: Router
+  ) {}
+
+  /*
+  @Usage  This property holds the type safe form group fields for basic information view.
+  */
+  basicInfoObj = new FormGroup<BasicInfo>({
+    firstName: new FormControl(''),
+    lastName: new FormControl(''),
+    salutation: new FormControl('mr'),
+  });
+
+  /*
+  @Usage In this method route change is observed and handling is done.
+  */
+  ngOnInit(): void {
+    this.totalSteps = this.sideMenuList.length;
+    this.progressPercentage = 0;
+    this.sideMenuList[0].isActive = true;
+    this.currentMenu = this.sideMenuList[0];
+
+    const subscription = this.activeRoute.queryParams.subscribe((params) => {
+      const step = Number(params.step);
+      if (step > 0 && step <= this.totalSteps) {
+        this.sideMenuList.filter((obj) => (obj.isActive = false));
+        this.currentMenu = this.sideMenuList.filter(
+          (obj) => obj.id === Number(params.step)
+        )[0];
+        this.currentMenu.isActive = true;
+      } else {
+        console.warn('Invalid query param');
+        this.sideMenuList[0].isActive = true;
+        this.currentMenu = this.sideMenuList[0];
+      }
+    });
+    this.subscriptions.push(subscription);
+  }
+
+  /*
+  @Usage In this method navigation to next step is handled.
+  */
+  navigateToNextForm(): void {
+    if (this.currentMenu.id !== this.totalSteps) {
+      const nextMenuIndex = this.sideMenuList.indexOf(this.currentMenu) + 1;
+      this.onFormStepsNavigation(this.sideMenuList[nextMenuIndex]);
+    } else {
+      alert('Next is submit page which is yet to be implmented');
+    }
+  }
+
+  /*
+  @Usage In this method un subscribe event and restore to default values are handled.
+  */
+  ngOnDestroy(): void {
+    for (const subscription of this.subscriptions) {
+      subscription.unsubscribe();
+    }
+    this.basicInfoObj = new FormGroup<BasicInfo>({
+      firstName: new FormControl(''),
+      lastName: new FormControl(''),
+      salutation: new FormControl('mr'),
+    });
+  }
+
+  /*
+  @Usage In this method navigation to home page is handled.
+  */
+  onClose(): void {
+    this.navigationController.navigateForward(['/home']);
+  }
+
+  /*
+  @Usage In this method navigation to respective step is handled.
+  */
+  onFormStepsNavigation(event: {
+    id: number;
+    displayName: string;
+    isCompleted: boolean;
+    isActive: boolean;
+    selector: string;
+  }): void {
+    const param = event.id;
+    this.router.navigate([], {
+      relativeTo: this.activeRoute,
+      queryParams: {
+        step: param,
+      },
+    });
+  }
+
+  /*
+  @Usage In this methos progress values are updated.
+  */
+  pageProgressStatusCallBack(event: {
+    id: number;
+    displayName: string;
+    isCompleted: boolean;
+    isActive: boolean;
+    selector: string;
+  }): void {
+    console.log(event);
```

Comment:
Maybe, remove the console log?

Reply:
Ya I forgot it. May be we need to add a lint rule for this as well.